includes/class-newspack-newsletters-subscription.php

```diff
@@ -14,11 +14,31 @@ class Newspack_Newsletters_Subscription {
 
 	const API_NAMESPACE = 'newspack-newsletters/v1';
 
+	const EMAIL_VERIFIED_META    = 'newspack_newsletters_email_verified';
+	const EMAIL_VERIFIED_REQUEST = 'newspack_newsletters_email_verification_request';
+	const EMAIL_VERIFIED_CONFIRM = 'newspack_newsletters_email_verification';
+
+	const WC_ENDPOINT         = 'newsletters';
+	const SUBSCRIPTION_UPDATE = 'newspack_newsletters_subscription';
+
 	/**
 	 * Initialize hooks.
 	 */
 	public static function init() {
 		add_action( 'rest_api_init', [ __CLASS__, 'register_api_endpoints' ] );
+
+		/** User email verification for subscription management. */
+		add_action( 'resetpass_form', [ __CLASS__, 'set_current_user_email_verified' ] );
+		add_action( 'password_reset', [ __CLASS__, 'set_current_user_email_verified' ] );
```

Comment:
Shouldn't the email be set as verified also when the user is authenticated via magic link (`newspack_magic_link_authenticated` hook) or registered (`newspack_registered_reader` hook)?